src/ai/ai-brain.ts

```diff
@@ -3,43 +3,87 @@ import { KeyCode } from "../game/key-codes";
 import { Player } from "../game/player";
 import { Vector } from "../utility/vector";
 import { AiPoweredController } from "./ai-controller";
+import { isCrashImminent, sanitizeAngle } from "../utility/math";
 
 export class AiBrain {
     private goForwardTimeout = 0;
     private shootTimeout = 0;
     private MAX_GO_FORWARD_TIMEOUT = 2;
     private MAX_SHOOT_DELAY = 5;
     private MIN_SHOOT_DELAY = 2;
+    private state = "start"; // | tracking | evading | drifting
```

Comment:
There is some typescript construct to enforce typecheck on values assigned to this variable, but I can't remember how it is called or what's the syntax.

Reply:
You can create a string union type
```ts
type AiState = "A" | "B" | "C";
let state: State;
```
or derive the type from a const expression:
```ts
const aiStates = ["A" | "B" | "C"] as const;
type AiState = typeof stateArr[number];
let state: State"
```
But in this case, I'd go with an enum
```ts
enum AiState {
  Start, Tracking, Evading, Drifting
}
let State: AiState;
```